repository/state/state.go

```diff
@@ -264,7 +268,17 @@ func (st *State) Merge(stateID objects.Checksum, deltaState *State) {
 			subpart.Length,
 		)
 	}
-	deltaState.muSignatures.Unlock()
```

Comment:
is it on purpose?

Reply:
absolutely not, I created the pull request on the wrong commit, this got fixed in a commit I didn't include in the PR to split the diff. sorry, closing the PR and will reopen with the this fix and the int vs uint64 fix that Copilot choked upon.